zcash_client_backend/src/decrypt.rs

```diff
@@ -103,13 +103,36 @@ impl<A> DecryptedOutput<orchard::note::Note, A> {
 
 /// Scans a [`Transaction`] for any information that can be decrypted by the set of
 /// [`UnifiedFullViewingKey`]s.
+///
+/// # Parameters
+/// - `params`: The network parameters corresponding to the network the transaction
+///   was created for.
+/// - `mined_height`: The height at which the transaction was mined, or `None` for
+///   unmined transactions.
+/// - `chain_tip_height`: The current chain tip height, if known. This parameter
+///   will be unused if `mined_height.is_some()`.
+/// - `tx`: The transaction to decrypt.
+/// - `ufvks`: The [`UnifiedFullViewingKey`]s to use in trial decryption, keyed
+///   by the identifiers for the wallet accounts they correspond to.
 pub fn decrypt_transaction<'a, P: consensus::Parameters, AccountId: Copy>(
     params: &P,
-    height: BlockHeight,
+    mined_height: Option<BlockHeight>,
+    chain_tip_height: Option<BlockHeight>,
     tx: &'a Transaction,
     ufvks: &HashMap<AccountId, UnifiedFullViewingKey>,
 ) -> DecryptedTransaction<'a, AccountId> {
-    let zip212_enforcement = zip212_enforcement(params, height);
+    let zip212_enforcement = zip212_enforcement(
+        params,
+        // Height is block height for mined transactions, and the "mempool height" (chain height + 1)
+        // for mempool transactions. We fall back to Sapling activation if we have no other
+        // information.
```

Comment:
Should we not fall back to a height after Canopy activation if we have no information? New wallets will have a birthday height after Canopy.

Reply:
Ah, I had misremembered how the spec (and hopefully the implementation?) handles decrypting a lead-byte 0x02 ciphertext before Canopy. It is accepted:
![image](https://github.com/user-attachments/assets/317a09fd-495e-405f-bb09-e6e1978f466c)
So this is "generous in what we accept", which is fine in this particular context.